docs/jumpstart/week-3.md

```diff
@@ -0,0 +1,418 @@
+        return;
+    }
+    console.log(num);
+}
+```
+
+### Functions as Data
+
+Functions in Javascript are treated very similarly to other types of data like objects or strings. You can assign a variable to a function itself, and you can even pass functions as arguments to other functions.
+
+```js
+function adder(num1, num2) {
+    return num1 + num2;
+}
+// Note that this is different than setting 'const adderFunc = adder()'
+// Because we ommit the parenthasis, we don't execute the function yet.
+const adderFunc = adder;
+
+function arrayCombiner(array, combinerFunc) {
+    let sum = 0;
+    for (const el of array) {
+        // Here we are calling the function that we were passed as an argument
+        sum = combinerFunc(sum, el);
+    }
+    return sum;
+}
+
+const numberArray = [1, 2, 3, 4, 5];
+const arraySum = arrayCombiner(numberArray, adderFunc);
+// Array sum is now equal to 15, the sum of every element of numberArray
+```
+
+We can also return functions from functions.
+
+```js
+function getAdder(num1, num2) {
+    return function adder() {
+        return num1 + num2;
+    }
+}
+
+// Here we give the arguments we want to add, but the actual addition is not executed yet
+const adderFunction = getAdder(3, 4);
+
+// Here the addition is executed and sum is equal to 7
+const sum = adderFunction();
+```
+
+### Anonymous Functions
+
+We'll be storing and passing functions around a lot in React, and it becomes tedious to have to write out `function someName()` every time. Furthermore, in our last example we returned a function named 'adder' but that function name is never used anywhere.
+
+To make things more convenient, Javascript defines a quick way to define functions without names.
+
+```js
+const myFunctionLong = function operator(num1, num2) {
+    const reverse = num1 * -1;
+    return reverse + num2;
+}
+
+// This syntax is equivielent to above
+const myFunctionQuick = (num1, num2) => {
+    const reverse = num1 * -1;
+    return reverse + num2;
+}
+```
+
+An anonymous function drops the `function` keyword and the function name. Instead, we just define the argument list between two parenthasis and use an arrow (`=>`) pointing to the function block.
+
+If our function doesn't have any arguments we just use empty parentheses: `() => { ... }`
+
+We can shorten this syntax even further if we don't have any intermediate steps to preform before returning a value. If we drop the `{}` and point directly to a code statement, and the anonymous function will return that statement as if you had specified it with a `return` keyword.
+
+```js
+// These syntaxes are all equivielent!
+
+const fiveReturner = function fiveReturnerInner() {
+    return 'five';
+}
+
+const fiveReturnerFast = () => {
+    return 'five';
+}
+
+const fiveReturnerFastest = () => 'five';
+```
+
+
+## Additional Resources
+
+For a great interactive tutorial through the basics of Javascript try [Codecademy](https://www.codecademy.com/learn/introduction-to-javascript).
+
```

Comment:
Maybe throw this down there too https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference